azurelinuxagent/common/protocol/wire.py

```diff
@@ -793,20 +793,55 @@ def get_ext_conf(self):
         return self.ext_conf
 
     def get_ext_manifest(self, ext_handler, goal_state):
-        local_file = MANIFEST_FILE_NAME.format(ext_handler.name,
-                                               goal_state.incarnation)
-        local_file = os.path.join(conf.get_lib_dir(), local_file)
-        xml_text = self.fetch_manifest(ext_handler.versionUris)
-        self.save_cache(local_file, xml_text)
-        return ExtensionManifest(xml_text)
+        update_goal_state = False
+
+        while True:
+            try:
+                if update_goal_state:
+                    update_goal_state = False
```

Comment:
should this be set after the update? otherwise a bad request there would just keep trying. 

the while true makes me twitchy, I'm not sure by reading the method what the exit criteria would be.